### PR TITLE
Linear_Search Algorithm in Kotlin

### LINKED ISSUE #1113: Algorithms of Linear_Search, Selection_Sort, Insertion_Sort and Binary_Search in Kotlin.
I want to write linear search, selection sort, insertion sort and binary search in kotlin.
@jainaman224 please assign it to me.

### PR DESCRIPTION
**Pull request by shubhammantri1**
Fixes #1113  

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Linear Search Algorithm in Kotlin
- Insertion Sort Algorithm in Kotlin
- Selection Sort Algorithm in Kotlin

#### Languages Used:
- Kotlin 

#### Files Added:
- Linear_Search.kt
- Insertion_Sort.kt
- Selection_Sort.kt

### CODE REVIEW

**Comment by shubhammantri1** [comment]
Please Review My Algorithm @rishabhgarg25699

File: Linear_Search/Linear_Search.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
+//Function For Linear Search
+fun Linear_Search(list: List<Int>, Num: Int):Int{
+    //If number not found than return -1 otherwise return postion of that number.
+    var foundAt:Int = -1
+    for(number in 0 until list.size){
+        if(list[number]==Num){
+            foundAt = number
+
```
Comment (amaaniqbal) [inline Linear_Search/Linear_Search.kt]: Remove unnecessary spaces

File: Linear_Search/Linear_Search.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
+//Function For Linear Search
+fun Linear_Search(list: List<Int>, Num: Int):Int{
+    //If number not found than return -1 otherwise return postion of that number.
+    var foundAt:Int = -1
+    for(number in 0 until list.size){
```
Comment (amaaniqbal) [inline Linear_Search/Linear_Search.kt]: Add a space between ')' and '{'

File: Linear_Search/Linear_Search.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        if(list[number]==Num){
```
Comment (amaaniqbal) [inline Linear_Search/Linear_Search.kt]: Give spacings between the operators and in between the operator and operand everywhere.

File: Linear_Search/Linear_Search.kt (hunk 2 above)
Comment (shubhammantri1) [reply Linear_Search/Linear_Search.kt]: Done @amaaniqbal

File: Linear_Search/Linear_Search.kt (hunk 3 above)
Comment (shubhammantri1) [reply Linear_Search/Linear_Search.kt]: Done

File: Linear_Search/Linear_Search.kt (hunk 2 above)
Comment (shubhammantri1) [reply Linear_Search/Linear_Search.kt]: Done

File: Linear_Search/Linear_Search.kt (hunk 1 above)
Comment (shubhammantri1) [reply Linear_Search/Linear_Search.kt]: Done @amaaniqbal

File: Linear_Search/Linear_Search.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+//Driver Function For Linear Search
+fun main(args:Array<String>){
+    val list = listOf(2, 7, 10, 45, 60, 5)
+    val position: Int
+    //Let 45 to be searched
+    position = Linear_Search(list, 45)
+
+    if(position==-1){
+      println("Number Not Found")
```
Comment (somya-kapoor) [inline Linear_Search/Linear_Search.kt]: Please fix the indentation.

File: Linear_Search/Linear_Search.kt [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    //Now search for 1
+    val position1: Int
+    position1 = Linear_Search(list, 1)
+    if(position1==-1){
+        println("Number Not Found")
+    }
+    else{
+        println("Number Found at position ${position1+1}")
+    }
+//Output:-
```
Comment (somya-kapoor) [inline Linear_Search/Linear_Search.kt]: Add a new line before output

Comment (shubhammantri1) [reply Linear_Search/Linear_Search.kt]: Done

File: Linear_Search/Linear_Search.kt (hunk 2 above)
Comment (amaaniqbal) [reply Linear_Search/Linear_Search.kt]: I still see them at quite a few places. Line 5, 19, 28, 31 etc.

Comment (shubhammantri1) [reply Linear_Search/Linear_Search.kt]: Done

File: Linear_Search/Linear_Search.kt [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        println("Number Not Found")
+    }
+    else {
+        println("Number Found at position ${position1 + 1}")
+    }
+}
+
+//Output:-
+// Number Found at position 4
+// Number Not Found
```
Comment (MastersAbh) [inline Linear_Search/Linear_Search.kt]: Add a blank line at the end of the file.

File: Linear_Search/Linear_Search.kt [hunk 7]
```diff
@@ -0,0 +1,38 @@
+//Function For Linear Search
+fun Linear_Search(list: List<Int>, Num: Int):Int{
```
Comment (MastersAbh) [inline Linear_Search/Linear_Search.kt]: Add a space before {

Comment (shubhammantri1) [reply Linear_Search/Linear_Search.kt]: Done

File: Linear_Search/Linear_Search.kt [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+//Driver Function For Linear Search
+fun main(args:Array<String>) {
+    val list = listOf(2, 7, 10, 45, 60, 5)
+    val position: Int
+    //Let 45 to be searched
+    position = Linear_Search(list, 45)
+    if(position == -1) {
+      println("Number Not Found")
```
Comment (MastersAbh) [inline Linear_Search/Linear_Search.kt]: Follow 4 space indentation here.

Comment (MastersAbh) [inline Linear_Search/Linear_Search.kt]: Follow 4 space indentation here.

Comment (shubhammantri1) [reply Linear_Search/Linear_Search.kt]: Done

**Comment by shubhammantri1** [comment]
I added Insertion Sort Algorithm. Please Review it @jainaman224

File: Insertion_Sort/Insertion_Sort.kt [hunk 9]
```diff
@@ -0,0 +1,26 @@
+//Function for sorting an list using Insertion Sort
+fun Insertion_sort(list: MutableList<Int>): List<Int> {
+    for(i in 0 until list.size) {
+            var key = list[i]
+            var j = i - 1
```
Comment (MastersAbh) [inline Insertion_Sort/Insertion_Sort.kt]: why 8 spaces instead of 4 here ?

File: Insertion_Sort/Insertion_Sort.kt [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+//Driver function for Insertion sort
+fun main(args: Array<String>) {
+    val list = mutableListOf(4, 67, 89, 2, 45, 102)
+    println("Unsorted List: $list")
+    val sortedlist = Insertion_sort(list)
+    println("Sorted List: $sortedlist")
+}
+
+//Input: Unsorted: [4, 67, 89, 2, 45, 102]
+//Output: Sorted: [2, 4, 25, 67, 89, 102]
```
Comment (MastersAbh) [inline Insertion_Sort/Insertion_Sort.kt]: Add a blank line at the end of file

Comment (shubhammantri1) [reply Insertion_Sort/Insertion_Sort.kt]: Sorry, But I solved that.

**Comment by shubhammantri1** [comment]
Adding Selection_Sort. Please Review it.

**Comment by jainaman224** [comment]
Don't add more files in this PR.
Create separate PR for each new algorithm.

File: Selection_Sort/Selection_Sort.kt [hunk 11]
```diff
@@ -0,0 +1,25 @@
+//Function for Selection Sort
+fun Selection_Sort(list: MutableList<Int>):List<Int> {
+    for(i in 0 until (list.size) ) {
+        var min_index: Int = i
+            for(j in (i+1) until (list.size)) {
```
Comment (jainaman224) [inline Selection_Sort/Selection_Sort.kt]: Fix indentation.

File: Selection_Sort/Selection_Sort.kt [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+                    min_index = j
+                }
+            }
+        var temp: Int = list[min_index]
+        list[min_index] = list[i]
+        list[i] = temp
+    }
+    return list
+}
+//Driver function for selection sort
```
Comment (jainaman224) [inline Selection_Sort/Selection_Sort.kt]: Add a new line above this.

File: Selection_Sort/Selection_Sort.kt (hunk 11 above)
Comment (shubhammantri1) [reply Selection_Sort/Selection_Sort.kt]: Done

File: Selection_Sort/Selection_Sort.kt (hunk 12 above)
Comment (shubhammantri1) [reply Selection_Sort/Selection_Sort.kt]: Done

**Pull request merged**